python/lsst/webcommon/decoders.py

```diff
@@ -0,0 +1,105 @@
+# LSST Data Management System
+# Copyright 2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+Decoders for type-based decoding for python objects.
+@author  Brian Van Klaveren, SLAC
+"""
+
+import json
+from .response import ScalarResponse, VectorResponse, ErrorResponse
+
+class TypeDecoder(json.JSONDecoder):
+    """
+    A JSONDecoder which allows subclasses to notify if they can decode a type.
+    """
+    def __init__(self, **kwargs):
+        super(TypeDecoder, self).__init__(object_hook=self.object_hook, **kwargs)
+
+    def decode_object(self, obj):
+        return obj
+
+    def can_decode(self, obj):
+        return False
+
+    def object_hook(self, obj):
+        if self.can_decode(obj):
+            return self.decode_object(obj)
+        return obj
+
+class ScalarResponseDecoder(TypeDecoder):
+    def __init__(self, **kwargs):
+        super(ScalarResponseDecoder, self).__init__(**kwargs)
+
+    def can_decode(self, obj):
+        return u"result" in obj.keys()
```

Comment:
Is `u` prefix needed here? If yes then for consistency add it to `obj["result"]` below